dev/com.ibm.ws.ejbcontainer/src/com/ibm/ws/ejbcontainer/osgi/internal/naming/LocalColonEJBNamingHelperImpl.java

```diff
@@ -70,25 +74,42 @@ protected void unsetEJBHomeRuntime(ServiceReference<EJBHomeRuntime> ref) {
     }
 
     @Override
-    public synchronized void bind(EJBBinding binding, String name) {
+    public synchronized boolean bind(EJBBinding binding, String name, boolean isSimpleName) {
         final boolean isTraceOn = TraceComponent.isAnyTracingEnabled();
+        boolean notAmbiguous = true;
+        EJBBinding localBinding = new EJBBinding(binding.homeRecord, binding.interfaceName, binding.interfaceIndex, binding.isLocal);
+
         if (isTraceOn && tc.isEntryEnabled()) {
             Tr.entry(tc, "bind: " + name);
         }
 
+        EJBBinding previousBinding = localColonEJBBindings.get(name);
+
+        // There won't be a previous binding for an ambiguous simple binding name
+        if (isSimpleName) {
+            localBinding.setAmbiguousReference();
+            notAmbiguous = false;
+        }
+
+        if (previousBinding != null) {
+            localBinding.setAmbiguousReference();
+            localBinding.addJ2EENames(previousBinding.getJ2EENames());
+            notAmbiguous = false;
+        }
+
         Lock writeLock = javaColonLock.writeLock();
         writeLock.lock();
         try {
-
-            //TODO: If LocalColonEJBBindings already contains name, bind ambiguous reference exception
-            localColonEJBBindings.put(name, binding);
+            localColonEJBBindings.put(name, localBinding);
         } finally {
             writeLock.unlock();
         }
 
         if (isTraceOn && tc.isEntryEnabled()) {
-            Tr.exit(tc, "bind");
+            Tr.exit(tc, "bind: " + notAmbiguous);
```

Comment:
Same here, if I saw just this in trace `bind: false` I would think the bind failed, not that it's not Ambiguous so I'd make this clearer.

Reply:
Updated exit message to "bind: notAmbiguous = "